From: "Dr. Tobias Quathamer" <toddy@debian.org>
Date: Wed, 31 Jan 2018 23:27:37 +0100
Subject: Prefer xdg-open

Add xdg-open as the first choice whenever this seems to be useful.
---
 lib/configure.py | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/lib/configure.py b/lib/configure.py
index 7fee68d..abc4d53 100644
--- a/lib/configure.py
+++ b/lib/configure.py
@@ -657,7 +657,7 @@ def checkFormatEntries(dtl_tools):
     checkViewerEditor('a Dia viewer and editor', ['dia'],
         rc_entry = [r'\Format dia        dia     DIA                    "" "%%"	"%%"	"vector,zipped=native", "application/x-dia-diagram"'])
     #
-    checkViewerEditor('an OpenDocument drawing viewer and editor', ['libreoffice', 'lodraw', 'ooffice', 'oodraw', 'soffice'],
+    checkViewerEditor('an OpenDocument drawing viewer and editor', ['xdg-open', 'libreoffice', 'lodraw', 'ooffice', 'oodraw', 'soffice'],
         rc_entry = [r'\Format odg        "odg, sxd" "OpenDocument drawing"   "" "%%"	"%%"	"vector,zipped=native"	"application/vnd.oasis.opendocument.graphics"'])
     #
     checkViewerEditor('a Grace viewer and editor', ['xmgrace'],
@@ -682,16 +682,16 @@ def checkFormatEntries(dtl_tools):
 \Format xbm        xbm     XBM                    "" "%s"	"%s"	""	"image/x-xbitmap"
 \Format xpm        xpm     XPM                    "" "%s"	"%s"	""	"image/x-xpixmap"'''
     path, iv = checkViewerNoRC('a raster image viewer',
-        ['xv', 'gwenview', 'kview',
+        ['xdg-open', 'xv', 'gwenview', 'kview',
          'eog', 'xviewer', 'ristretto', 'gpicview', 'lximage-qt',
-         'xdg-open', 'gimp-remote', 'gimp'],
+         'gimp-remote', 'gimp'],
         rc_entry = [imageformats])
     path, ie = checkEditorNoRC('a raster image editor',
         ['gimp-remote', 'gimp'], rc_entry = [imageformats])
     addToRC(imageformats % ((iv, ie)*10))
     #
     checkViewerEditor('a text editor',
-        ['xemacs', 'gvim', 'kedit', 'kwrite', 'kate',
+        ['xdg-open', 'xemacs', 'gvim', 'kedit', 'kwrite', 'kate',
          'nedit', 'gedit', 'geany', 'leafpad', 'mousepad', 'xed', 'notepad'],
         rc_entry = [r'''\Format asciichess asc    "Plain text (chess output)"  "" ""	"%%"	""	""
 \Format docbook    sgml    DocBook                B  ""	"%%"	"document,menu=export"	""
@@ -745,7 +745,7 @@ def checkFormatEntries(dtl_tools):
     #checkProg('a Postscript interpreter', ['gs'],
     #  rc_entry = [ r'\ps_command "%%"' ])
     checkViewer('a Postscript previewer',
-                ['kghostview', 'okular', 'qpdfview --unique',
+                ['xdg-open', 'kghostview', 'okular', 'qpdfview --unique',
                  'evince', 'xreader',
                  'gv', 'ghostview -swap', 'gsview64', 'gsview32'],
         rc_entry = [r'''\Format eps        eps     EPS                    "" "%%"	""	"vector"	"image/x-eps"
@@ -756,7 +756,7 @@ def checkFormatEntries(dtl_tools):
     # maybe use "bestApplication()" from https://github.com/jleclanche/python-mime
     # the MIME type is set for pdf6, because that one needs to be autodetectable by libmime
     checkViewer('a PDF previewer',
-                ['pdfview', 'kpdf', 'okular', 'qpdfview --unique',
+                ['xdg-open', 'pdfview', 'kpdf', 'okular', 'qpdfview --unique',
                  'evince', 'xreader', 'kghostview', 'xpdf', 'SumatraPDF',
                  'acrobat', 'acroread', 'mupdf',
                  'gv', 'ghostview', 'AcroRd32', 'gsview64', 'gsview32'],
